controls/SV-244546.rb

```diff
@@ -18,114 +20,86 @@
 
-    permissive = 1
+Once it is determined the whitelist is built correctly, set the fapolicyd to enforcing mode by editing the "permissive" line in the /etc/fapolicyd/fapolicyd.conf file.
 
-    Build the whitelist in the \"/etc/fapolicyd/fapolicyd.rules\" file ensuring
-the last rule is \"deny perm=any all : all\".
-
-    Once it is determined the whitelist is built correctly, set the fapolicyd
-to enforcing mode by editing the \"permissive\" line in the
-/etc/fapolicyd/fapolicyd.conf file.
-
-    permissive = 0
-  "
+permissive = 0'
   impact 0.5
   tag severity: 'medium'
   tag gtitle: 'SRG-OS-000368-GPOS-00154'
-  tag satisfies: ['SRG-OS-000368-GPOS-00154', 'SRG-OS-000370-GPOS-00155',
-'SRG-OS-000480-GPOS-00232']
+  tag satisfies: ['SRG-OS-000368-GPOS-00154', 'SRG-OS-000370-GPOS-00155', 'SRG-OS-000480-GPOS-00232']
   tag gid: 'V-244546'
-  tag rid: 'SV-244546r743887_rule'
+  tag rid: 'SV-244546r858730_rule'
   tag stig_id: 'RHEL-08-040137'
-  tag fix_id: 'F-47778r743886_fix'
+  tag fix_id: 'F-47778r858729_fix'
   tag cci: ['CCI-001764']
   tag nist: ['CM-7 (2)']
 
-  if virtualization.system.eql?('docker')
+  if !input('use_fapolicyd')
+    impact 0.0
+    describe 'The organization is not using the Fapolicyd service to manage firewall servies, this control is Not Applicable' do
+      skip 'The organization is not using the Fapolicyd service to manage firewall servies, this control is Not Applicable'
+    end
+  elsif virtualization.system.eql?('docker')
     impact 0.0
-    describe "Control not applicable within a container" do
-      skip "Control not applicable within a container"
+    describe 'Control not applicable within a container' do
+      skip 'Control not applicable within a container'
     end
   else
     describe parse_config_file('/etc/fapolicyd/fapolicyd.conf') do
-      its('permissive') { should eq 0 }
-    end
-  
-    describe file('/etc/fapolicyd/fapolicyd.rules') do
-      it { should exist }
+      its('permissive') { should cmp 0 }
     end
-  
-    describe file('/etc/fapolicyd/fapolicyd.rules').content.strip.split("\n")[-1] do
-      it { should cmp 'deny all all' }
-    end if file('/etc/fapolicyd/fapolicyd.rules').exist?
 
-    system_mounts = command("mount | egrep '^tmpfs| ext4| ext3| xfs' | awk '{ printf \"%s\\n\", $3 }'").stdout.split
+    rules_file = '/etc/fapolicyd/compiled.rules'
+    rules_file = '/etc/fapolicyd/fapolicyd.rules' if os.release.to_f < 8.4
 
-    describe file('/etc/fapolicyd/fapolicyd.mounts') do
+    describe file(rules_file) do
       it { should exist }
     end
-  
-    describe file('/etc/fapolicyd/fapolicyd.mounts') do
-      its('content.split') { should match_array system_mounts }        
-    end if file('/etc/fapolicyd/fapolicyd.mounts').exist?
+
+    if file(rules_file).exist?
+      describe file(rules_file).content.strip.split("\n")[-1] do
+        it { should cmp 'deny perm=any all : all' }
+      end
```

Comment:
@wdower @em-c-rod we should verify that this failure is not perhaps a typo in the guidance.
```suggestion
Verify this is not a typo in the guidance, given this is how the Ansible Lockdown hardening configures it this way for v1r12
```ruby
  expected: deny perm=any all : all
  got: deny all all```
```